packages/admin/cms-admin/src/userPermissions/user/permissions/selectScopesDialog/SelectScopesDialog.tsx

```diff
@@ -0,0 +1,135 @@
+import { useApolloClient } from "@apollo/client";
+import {
+    CancelButton,
+    DataGridToolbar,
+    Field,
+    FinalForm,
+    SaveBoundary,
+    SaveBoundarySaveButton,
+    ToolbarFillSpace,
+    ToolbarItem,
+    useFormApiRef,
+} from "@comet/admin";
+import { Dialog, DialogActions, DialogTitle } from "@mui/material";
+import { DataGrid, GridColDef, GridToolbarQuickFilter } from "@mui/x-data-grid";
+import gql from "graphql-tag";
+import { FormattedMessage, useIntl } from "react-intl";
+
+import { generateGridColumnsFromContentScopeProperties } from "../ContentScopeGrid";
+import { GQLContentScopesQuery } from "../ContentScopeGrid.generated";
+import { GQLUpdateContentScopesMutation, GQLUpdateContentScopesMutationVariables } from "./SelectScopesDialog.generated";
+
+export interface SelectScopesDialogProps {
+    open: boolean;
+    onClose: () => void;
+    data: GQLContentScopesQuery;
```

Comment:
The dialog is now the data structure defined by the parent. A better approach would be to inverse this: The dialog defines a fragment with the data it needs, which is then loaded by the parent component.

Reply:
Yes, even better!

Reply:
so basically for the initial State of the form i pass the userContentScopes as a prop as well as the userContentScopesSkipManual for disabling in the Datagrid but do a query in the dialog for loading the availableContentScopes ?

Reply:
Either this, or load them in the dialog again. Should be loaded from the cache anyway.